app/helpers/admin/analytics_helper.rb

```diff
@@ -67,25 +67,78 @@ def total_number_of_searches
       total_searches.html_safe
     end
 
+    def avg_number_searches_per_visit
+      avg_searches_per_visit =
+        "#{AhoyQueries.get_total_number_of_searches
+        .div(AhoyQueries.get_total_number_of_ahoy_visits)}"
+      avg_searches_per_visit.html_safe
+    end
+
     def number_of_visits_initiated_from_homepage
       visits_initiated_from_homepage =
         "#{AhoyQueries.get_number_of_visits_initiated_from_homepage}"
       visits_initiated_from_homepage.html_safe
     end
 
-    def most_visited_locations
-      AhoyQueries.get_most_visited_locations(5)
-    end
-
     def most_used_keywords
       AhoyQueries.get_most_used_keywords(10)
     end
 
-    def avg_number_searches_per_visit
-      avg_searches_per_visit =
-        "#{AhoyQueries.get_total_number_of_searches
-        .div AhoyQueries.get_total_number_of_ahoy_visits}"
-      avg_searches_per_visit.html_safe
+    # calculates the average rating for the given keyword (includes only rated searches)
+    def get_average_search_rating(keyword)
+      all_rated_search_events = AhoyQueries.get_all_rated_search_events_for_keyword(keyword)
+
+      sum_ratings = all_rated_search_events.reduce(0) { |sum, num| sum + num.properties['rating'] }
+
+      number_of_rated_events = all_rated_search_events.length
+      average_rating =
+        if number_of_rated_events.zero?
+          0
+        else
+          sum_ratings/number_of_rated_events
+        end
+
+      [average_rating, number_of_rated_events]
+    end
+
+    def number_of_searches_for_rating(rating)
+      total_searches_for_rating =
+        "(#{AhoyQueries.get_number_of_searches_for_rating(rating)} ratings)"
+
+      total_searches_for_rating.html_safe
+    end
+
+    # retrieves all the search events for the given rating
+    def searches_for_rating(rating)
+      searches_with_rating = AhoyQueries.get_search_events_for_rating(rating)
+
+      group_keywords_and_count(searches_with_rating)
+    end
+
+    def keyword_search_rating_details(keyword, rating)
+      keyword_searches_with_rating = AhoyQueries.get_keyword_search_rating_details(keyword, rating)
+
+      keyword_search_breakdown = {}
+      keyword_searches_with_rating.each do |search|
+        main_category = search.properties['main_category']
```

Comment:
```suggestion
        main_category = search.properties['main_category'] || "--"
```
this will eliminate the next line

Reply:
@anna-dorigo unfortunately it wouldn't work as intended because `properties['main_category']` always exists in the hash. What happens is that it will take the empty string if we try to do it in one line and never the `--` characters.

Reply:
I see my bad

Reply:
No problem, thank you for confirming.